pandas/core/frame.py

```diff
@@ -3321,8 +3325,7 @@ def sort_index(self, axis=0, level=None, ascending=True, inplace=False,
         axis = self._get_axis_number(axis)
         labels = self._get_axis(axis)
 
-        # sort by the index
-        if level is not None:
```

Comment:
yes in fact this was what prevented

``.sort_index()`` and ``.sort_index(level=0)`` from being the same.